File: app/src/main/java/langchain4j/BlueSky.java
package langchain4j;

import dev.langchain4j.model.chat.ChatModel;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class BlueSky {

  public static void main(String[] args) {
    ChatModel model = ChatModelFactory.getChatModel(ChatModelProvider.OPENAI);

    // Send a user message and get the response
    String response = model.chat("Why is the sky blue?");

    log.info(response);
  }
}

File: app/src/main/kotlin/langchain4j/BlueSky.kt
package langchain4j

import dev.langchain4j.model.chat.ChatModel
import dev.langchain4j.model.openai.OpenAiChatModel
import dev.langchain4j.model.openai.OpenAiChatModelName.GPT_5_NANO
import mu.KotlinLogging

private val log = KotlinLogging.logger {}

fun main() {
  val model: ChatModel =
      OpenAiChatModel.builder()
          .apiKey(System.getenv("OPENAI_API_KEY"))
          .modelName(GPT_5_NANO)
          .build()

  // Send a user message and get the response
  val response = model.generate("Why is the sky blue?")

  log.info { response }
}
